src/main/java/com/google/jenkins/plugins/k8sengine/client/ClientFactory.java

```diff
@@ -171,4 +171,15 @@ public String getDefaultProjectId() {
   public String getCredentialsId() {
     return this.credentialsId;
   }
+
+  /**
+   * Get access token for service account with this credentialsId.
+   *
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws IOException If an error occurred fetching the access token.
+   */
+  public String getAccessToken() throws IOException {
```

Comment:
This breaks the orthogonality of this class, as its no longer just a client factory. I suggest stripping out the credential initialization logic into its own class which can be invoked in this class's constructor as as needed elsewhere.

Reply:
I'm confused, when you say orthogonality, do you mean isolation?

Reply:
I moved the logic into the CredentialsUtil class. Let me know what you think.